locations/storefinders/storefrontgateway.py

```diff
@@ -3,20 +3,28 @@
 from scrapy import Spider
 from scrapy.http import JsonRequest, Response
 
+from locations.automatic_spider_generator import AutomaticSpiderGenerator, DetectionRequestRule, DetectionResponseRule
 from locations.dict_parser import DictParser
 from locations.hours import OpeningHours
 from locations.items import Feature
 from locations.pipelines.address_clean_up import clean_address
 
 
-class StorefrontgatewaySpider(Spider):
+class StorefrontgatewaySpider(Spider, AutomaticSpiderGenerator):
     """
     A relatively unknown storefinder, associated with https://mi9retail.com/
 
     To use, specify `start_urls`
     """
 
     start_urls = []
+    api_key: str = ""
+    detection_rules = [
+        DetectionRequestRule(url=r"^https?:\/\/storefrontgateway\.[w\.-]+/api/stores"),
```

Comment:
The last part of the regex doesn't look right from a quick look, for example, the `w` in the regex class is probably meant to be `\w` as a character class and the hyphen in the regex class is incorrect and probably meant to be escaped. Would the following work: `r"^https?:\/\/storefrontgateway(?:\.[\w\-]+)+\/api\/stores\/?$"`   Still not great for domain matching as it'll match invalid DNS names still, but it's not too bad of an approximation.

Reply:
Oops just occurred to be the named PCRE capture group is missing. Suggestion is then:

`DetectionRequestRule(url=r"^(?P<start_urls__list>https?:\/\/storefrontgateway(?:\.[\w\-]+)+\/api\/stores)\/?$")`